### PR TITLE
[AN-4508] Adds new GIF button to cursor menu

### PR DESCRIPTION
**Pull request by shumengye**
#### Description
Works with edge backend

Note! Pre-filling Giphy search with text from input is not implemented yet

#### Ticket
https://wearezeta.atlassian.net/browse/AN-4508
#### APK
[Download build #7595](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7595/artifact/build/artifact/wire-dev-PR160-7595.apk)
[Download build #7597](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7597/artifact/build/artifact/wire-dev-PR160-7597.apk)
[Download build #7598](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7598/artifact/build/artifact/wire-dev-PR160-7598.apk)
[Download build #7601](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7601/artifact/build/artifact/wire-dev-PR160-7601.apk)
[Download build #7602](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7602/artifact/build/artifact/wire-dev-PR160-7602.apk)
[Download build #7603](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7603/artifact/build/artifact/wire-dev-PR160-7603.apk)

### CODE REVIEW

**Comment by xuanzhaopeng** [comment]
Search "ghh" in gif

```java
Process: com.waz.zclient.dev, PID: 5790
                                                 java.lang.ArrayIndexOutOfBoundsException: src.length=44 srcPos=25 dst.length=44 dstPos=2 length=42
                                                     at java.lang.System.arraycopy(Native Method)
                                                     at android.support.v7.widget.StaggeredGridLayoutManager$LazySpanLookup.offsetForAddition(StaggeredGridLayoutManager.java:2717)
                                                     at android.support.v7.widget.StaggeredGridLayoutManager.handleUpdate(StaggeredGridLayoutManager.java:1482)
                                                     at android.support.v7.widget.StaggeredGridLayoutManager.onItemsAdded(StaggeredGridLayoutManager.java:1438)
                                                     at android.support.v7.widget.RecyclerView$6.dispatchUpdate(RecyclerView.java:748)
                                                     at android.support.v7.widget.RecyclerView$6.onDispatchSecondPass(RecyclerView.java:765)
                                                     at android.support.v7.widget.AdapterHelper.consumePostponedUpdates(AdapterHelper.java:122)
                                                     at android.support.v7.widget.AdapterHelper.consumeUpdatesInOnePass(AdapterHelper.java:549)
                                                     at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3022)
                                                     at android.support.v7.widget.RecyclerView.onMeasure(RecyclerView.java:2625)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.widget.LinearLayout.measureVertical(LinearLayout.java:847)
                                                     at android.widget.LinearLayout.onMeasure(LinearLayout.java:588)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:135)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1404)
                                                     at android.widget.LinearLayout.measureVertical(LinearLayout.java:695)
                                                     at android.widget.LinearLayout.onMeasure(LinearLayout.java:588)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5169)
                                                     at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
                                                     at com.android.internal.policy.impl.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2310)
                                                     at android.view.View.measure(View.java:16515)
                                                     at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:1958)
                                                     at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1155)
                                                     at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1337)
                                                     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1042)
                                                     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5721)
                                                     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
                                                     at android.view.Choreographer.doCallbacks(Choreographer.java:574)
                                                     at android.view.Choreographer.doFrame(Choreographer.java:544)
                                                     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
                                                     at android.os.Handler.handleCallback(Handler.java:733)
                                                     at android.os.Handler.dispatchMessage(Handler.java:95)
                                                  at android.os.Looper.loop(Loop
```

**Comment by shumengye** [comment]
build again

File: app/src/main/java/com/waz/zclient/controllers/giphy/GiphyController.java [hunk 1]
```diff
@@ -79,6 +86,11 @@ public void cancel() {
 
     @Override
     public boolean handleInput(@NonNull final String text, boolean afterPressedEnter) {
+        searchTrending();
+        return true;
+
+        // TODO: Support pre-filled search term
```
Comment (zbsz) [inline app/src/main/java/com/waz/zclient/controllers/giphy/GiphyController.java]: did you forget to fix that?

Comment (shumengye) [inline app/src/main/java/com/waz/zclient/controllers/giphy/GiphyController.java]: @zbsz was intentionally left as TODO, I've added that functionality now

**Comment by markesaaremets** [comment]
@shumengye 
- Kill the word-limit
- Just 1pt space between gifs, and not  at the edges
- Loading bar is shifting the other UI elements (for example icons on app bar)

**Pull request merged**